util/ubuntu-dependencies.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# Abort on any error.
+set -e
+
+# This script installs Streisand builder dependencies on an Ubuntu
+# 16.04 system; Debian jessie has been lightly tested. It installs
+# system-wide packages.
+#
+# It's safe to run this script multiple times.
+
+# We default to a regular, interactive upgrade:
+quiet=
+# quiet='--yes'
+
+# If you *really* want no interaction (for example, overwriting local
+# config files without prompting), then uncomment this:
+# export DEBIAN_FRONTEND=noninteractive
+
+# If you'd prefer to install dependencies in ~/.local/bin, uncomment
+# this function and comment the next.
+
+#function our_pip_install () {
+#    pip install --user --upgrade "$@"
+#}
+
+function our_pip_install () {
+    sudo -H pip install --upgrade "$@"
+}
+
+### No options below this line. ###
+
+sudo apt-get update
+sudo apt-get $quiet upgrade
+
+# We explicitly want word splitting.
+# shellcheck disable=SC2046
+sudo apt-get $quiet install $(cat <<EOF
+build-essential
+python-dev
+python-setuptools
+python-pip
+python-cffi libffi-dev
+EOF
+)
+
+# Debian doesn't have python-nacl.
```

Comment:
Again I think I would prefer to see Debian systems detected by lsb_release or some other mechanism and then have a case in-script that can take the required actions for Debian when its known to be required. I think longer term it will be safer than trying to install some packages and if it fails, installing some other packages.

Reply:
I'll do this as a matter of house style, since I don't care that much. I'll probably support (i.e., detect) mainline Debian and Ubuntu LTS. My reasoning for shotgun installs like that is: 

It's the capability I need, not knowledge of which flavor of Debian-derived system we're on. If I want to special-case by OS release or OS flavor, I need to put a big `case` statement at the top telling people, "Good luck, we don't have an installer for Devuan codename "Ascii", edit this script if you think I'm wrongheaded."

Maybe I'm hanging out with the Python people too much, who love to do "try: import foo; except: import somethingelse as foo".  Nowhere near as bad as Ruby's `rescue`, at least.

Reply:
That's fair, maybe I'm overthinking things. If you don't mind doing this for Debian and Ubuntu I would appreciate it!

Reply:
Done.